### ISSUE TITLE
TypeError: Cannot read property 'call' of undefined

### ISSUE THREAD

**Issue by simllll** [nuxt/nuxt.js#7918]
I have a kind of critical bug with "modern" build mode (not in dev, not in non modern mode).

TypeError: Cannot read property 'call' of undefined
    at r (3a431c40f1ac84aab470-app.js:formatted:41)
    at Object.693 (5e2083c4.js:1)
    at r (3a431c40f1ac84aab470-app.js:formatted:41)
    at Object.853 (0d43857a.js:1)
    at r (3a431c40f1ac84aab470-app.js:formatted:41)
    at Object.607 (288dc8e0.js:1)
    at r (3a431c40f1ac84aab470-app.js:formatted:41)
    at Module.1003 (index.js:1)
    at r (3a431c40f1ac84aab470-app.js:formatted:41)
    at async http:/localhost.at:3000/website/_assets/70df557e5c5f4faf5bbc-app.js:2

it seems it loads the legacly build version at one point,
    at Object.693 (5e2083c4.js:1)
which is: _assets/legacy-3aa82e6f50e494e1c6cd-node_modules/5e2083c4.js
and then it tries to load a file that actually exists, but somehow didn't make it into the webpack "resolver".
![image](https://user-images.githubusercontent.com/5757263/90244189-8dfa2980-de30-11ea-8e17-911ca5eef99a.png)
e[d] (d = 9)
![image](https://user-images.githubusercontent.com/5757263/90244227-a10cf980-de30-11ea-9ce3-e09f160656e4.png)
is here, but probably due to the legacy vs modern mixup something fails then.

here element 9 is missing
![image](https://user-images.githubusercontent.com/5757263/90244633-6fe0f900-de31-11ea-97d1-5449549375be.png)

### Versions

- nuxt: v2.14.1
- node: v12.18.3

### Reproduction

Pretty hard to track down right now

### What is Expected?
shouldn't fail with modern mode, it only happens when build mode modern is set.

### What is actually happening?
it fails to load a javascript file and the whole site stops working.

**Comment by simllll** [comment]
Not sure if relevant, but if I inspect the array of "e", there are mainly "modern" assets, but "693" (the point where it loasd the legacy file) is pointing to the legacy version. Are there any valid reasons that a legacy version is loaded? E.g. if there is no difference between a modern build?

**Comment by simllll** [comment]
It breaks here:
// EXTERNAL MODULE: ./node_modules/@babel/runtime/helpers/esm/classCallCheck.js
var classCallCheck = __webpack_require__(9);

and the really werid thing: only on desktop version, not on mobile. Super strange, how can this be related at all? (Guess maybe somethign isn't loaded on mobile which causes the issue, but not quite sure yet)

**Comment by pi0** [comment]
Hi. Asset names seem not matching 2.14. Are you using custom Chunk names? And any reproduce repo || URL?

**Comment by simllll** [comment]
hi @pi0, thanks for looking into it,
I have set custom filenames, but i removed this settings for now to track it down, doesn't change anything.
My setting was 
```js
	this.options.build.filenames = {
		...this.options.build.filenames,
		app: ({ isModern, isDev }) =>
			`${!isModern ? 'legacy-' : ''}${!isDev ? '[contenthash]' : '[name]'}-app.js`,
		chunk: ({ isModern, isDev }) =>
			`${!isModern ? 'legacy-' : ''}${!isDev ? '[contenthash]-' : ''}[name].js`
	};
```

Debug build is up and runnign at http://3d45e0f594df.ngrok.io/jobs

Current learnings:
* It works in mobile mode -> no idea why
* It can load something else from bubel/runtime, right before, but the classCallCheck.js fails
![image](https://user-images.githubusercontent.com/5757263/90253017-c9045900-de40-11ea-864c-2c847638e0c4.png)
* Its loaded through a date fn helper function, which I don't understand why it actually needs this one dependency at all (And therefore believe it already loads a wrong file?)

**Comment by simllll** [comment]
Regarding my last point:
according to the comments in the file it should load
```js
// EXTERNAL MODULE: ./node_modules/date-fns/esm/_lib/startOfUTCISOWeek/index.js
var startOfUTCISOWeek = __webpack_require__(693);
```
but it 693 is something else, it loads a compnent called HokSelect.vue instead.
```js

    /***/
    693: /***/
    (function(module, __webpack_exports__, __webpack_require__) {

        "use strict";

        // CONCATENATED MODULE: ./node_modules/vue-loader/lib/loaders/templateLoader.js??vue-loader-options!./node_modules/vue-loader/lib??vue-loader-options!./node_modules/@hokify/shared-components/lib/components/HokSelect.vue?vue&type=template&id=4fc308f6&scoped=true&
        var render = function() {
            var _vm = this;


```

**Comment by simllll** [comment]
I'm stuck, my guess is that something seems to be mixed up with modern and legacy build. But I'm out of ideas how to track this down. 

It should load id 693 from /website/_assets/node_modules/288dc8e0.f27f277.modern.js (the same file where the call of __webpack_require__(693) is happening, but it loads 

but it loads 693 instead from /website/_assets/node_modules/5e2083c4.4f8e4af.js which is a completely differnt code. The chunks seem to be mixed up. Module 693 is not in installedModules. The __webpack_require__ is the one from the "root", but the entry of 693 just points to a different function it seems.

**Comment by simllll** [comment]
I have one import to a component that is some times async and sometimes directly imported. I have changed it now that all occurences are imported async, and it seems this has somehow solved the issue. Still looking into it, not sure what is going on here though.

**Comment by clarkdo** [comment]
Hi @simllll , are you using ssr or spa mode? And what’s the modern mode are you using?

Can you send us the manifest files in your dist and also screenshot of the link and script tags in your broken page ?

**Comment by simllll** [comment]
Hi @clarkdo , thanks for your answer.

I'm using SSR, modern is set to "client".

manifest files:
[Uploading dist.zip…]()

desktop:
![image](https://user-images.githubusercontent.com/5757263/90265333-bb57cf00-de52-11ea-92dd-c46787e24ff0.png)
`<script nomodule src="/website/_assets/runtime.fa009b9.js" defer></script><script type="module" src="/website/_assets/runtime.3f02afd.modern.js" defer></script><script nomodule src="/website/_assets/40.c765e0f.js" defer></script><script type="module" src="/website/_assets/37.c2155fc.modern.js" defer></script><script nomodule src="/website/_assets/129.1dd27da.js" defer></script><script type="module" src="/website/_assets/126.7f1e543.modern.js" defer></script><script nomodule src="/website/_assets/30.86aeb8b.js" defer></script><script type="module" src="/website/_assets/288.16f3f35.modern.js" defer></script><script nomodule src="/website/_assets/130.be13865.js" defer></script><script type="module" src="/website/_assets/127.75b77f7.modern.js" defer></script><script nomodule src="/website/_assets/pages/jobs/index.f2a9af8.js" defer></script><script type="module" src="/website/_assets/pages/jobs/index.85d3d8e.modern.js" defer></script><script nomodule src="/website/_assets/node_modules/0d43857a.28504fe.js" defer></script><script type="module" src="/website/_assets/node_modules/0d43857a.cf4d41c.modern.js" defer></script><script nomodule src="/website/_assets/node_modules/288dc8e0.929319e.js" defer></script><script type="module" src="/website/_assets/node_modules/288dc8e0.f27f277.modern.js" defer></script><script nomodule src="/website/_assets/commons/62c6da1a.d8fb4f0.js" defer></script><script type="module" src="/website/_assets/commons/62c6da1a.de12725.modern.js" defer></script><script nomodule src="/website/_assets/pages/empty.8499b08.js" defer></script><script type="module" src="/website/_assets/pages/empty.4194d9e.modern.js" defer></script><script src="/website/_assets/node_modules/5e2083c4.4f8e4af.js" defer></script><script nomodule src="/website/_assets/42.bef11fd.js" defer></script><script type="module" src="/website/_assets/39.3b0d27a.modern.js" defer></script><script nomodule src="/website/_assets/278.fd24ac2.js" defer></script><script type="module" src="/website/_assets/274.0e6f0d0.modern.js" defer></script><script nomodule src="/website/_assets/14.9aaf416.js" defer></script><script type="module" src="/website/_assets/137.2902e3c.modern.js" defer></script><script nomodule src="/website/_assets/112.258d4b4.js" defer></script><script type="module" src="/website/_assets/11.cb95906.modern.js" defer></script><script nomodule src="/website/_assets/node_modules/commons.c1101f0.js" defer></script><script type="module" src="/website/_assets/node_modules/commons.b005655.modern.js" defer></script><script nomodule src="/website/_assets/app.2417406.js" defer></script><script type="module" src="/website/_assets/app.fa2606e.modern.js" defer></script>`

mobile:
![image](https://user-images.githubusercontent.com/5757263/90265410-d9bdca80-de52-11ea-95b3-9e7c228d6e28.png)
```

<script>!function(){var e=document,t=e.createElement("script");if(!("noModule"in t)&&"onbeforeload"in t){var n=!1;e.addEventListener("beforeload",function(e){if(e.target===t)n=!0;else if(!e.target.hasAttribute("nomodule")\|\|!n)return;e.preventDefault()},!0),t.type="module",t.src=".",e.head.appendChild(t),t.remove()}}();</script><script nomodule src="/website/_assets/runtime.fa009b9.js" defer></script><script type="module" src="/website/_assets/runtime.3f02afd.modern.js" defer></script><script nomodule src="/website/_assets/40.c765e0f.js" defer></script><script type="module" src="/website/_assets/37.c2155fc.modern.js" defer></script><script nomodule src="/website/_assets/129.1dd27da.js" defer></script><script type="module" src="/website/_assets/126.7f1e543.modern.js" defer></script><script nomodule src="/website/_assets/30.86aeb8b.js" defer></script><script type="module" src="/website/_assets/288.16f3f35.modern.js" defer></script><script nomodule src="/website/_assets/130.be13865.js" defer></script><script type="module" src="/website/_assets/127.75b77f7.modern.js" defer></script><script nomodule src="/website/_assets/pages/jobs/index.f2a9af8.js" defer></script><script type="module" src="/website/_assets/pages/jobs/index.85d3d8e.modern.js" defer></script><script nomodule src="/website/_assets/node_modules/0d43857a.28504fe.js" defer></script><script type="module" src="/website/_assets/node_modules/0d43857a.cf4d41c.modern.js" defer></script><script nomodule src="/website/_assets/node_modules/288dc8e0.929319e.js" defer></script><script type="module" src="/website/_assets/node_modules/288dc8e0.f27f277.modern.js" defer></script><script nomodule src="/website/_assets/commons/62c6da1a.d8fb4f0.js" defer></script><script type="module" src="/website/_assets/commons/62c6da1a.de12725.modern.js" defer></script><script nomodule src="/website/_assets/42.bef11fd.js" defer></script><script type="module" src="/website/_assets/39.3b0d27a.modern.js" defer></script><script nomodule src="/website/_assets/278.fd24ac2.js" defer></script><script type="module" src="/website/_assets/274.0e6f0d0.modern.js" defer></script><script nomodule src="/website/_assets/14.9aaf416.js" defer></script><script type="module" src="/website/_assets/137.2902e3c.modern.js" defer></script><script nomodule src="/website/_assets/112.258d4b4.js" defer></script><script type="module" src="/website/_assets/11.cb95906.modern.js" defer></script><script nomodule src="/website/_assets/node_modules/commons.c1101f0.js" defer></script><script type="module" src="/website/_assets/node_modules/commons.b005655.modern.js" defer></script><script nomodule src="/website/_assets/app.2417406.js" defer></script><script type="module" src="/website/_assets/app.fa2606e.modern.js" defer></script>
--
  | <script>window.addEventListener('load', function() { try{for(var nonCriticalCss=document.head.querySelectorAll("link[rel=stylesheet][media=none]"),i=0;i<nonCriticalCss.length;++i)nonCriticalCss[i].media="all"}catch(e){} })</script>


```

**Comment by clarkdo** [comment]
Thanks for the reporting information, I’ll take a look.

BTW, I suggest using server modern mode for SSR.

**Comment by simllll** [comment]
Unfortuantely our setup required client mode, as the web page is pre rendered in a warm cache. We already generating a version for mobile and a version for desktop, adding two more versions depending for each device ("modern" and "legacy") would need a lot of more ressources to fill and keep the cache fresh

**Comment by clarkdo** [comment]
From your information, I found <script src="/website/_assets/node_modules/5e2083c4.4f8e4af.js" defer>,  this file is legacy, so it should have a nomodule attr, but somehow it doesn’t and that’s why you’re seeing wrong chunk is imported.

I’ll have a deeper looking shortly.

**Comment by simllll** [comment]
> From your information, I found <script src="/website/_assets/node_modules/5e2083c4.4f8e4af.js" defer>, this file is legacy, so it should have a nomodule attr, but somehow it doesn’t and that’s why you’re seeing wrong chunk is imported.
> 
> I’ll have a deeper looking shortly.

Interesting, you are right, and this is also only on the desktop version which has the issues.

Any idea how to find out why this is included?

**Comment by simllll** [comment]
When I change some specfic imports to async, I found a working solution (hacky though), but jsut checked the output in this situation then (the filenames are different here again with my specific setup):
`<script nomodule src="/website/_assets/legacy-ba7a729375b1c859dd7b-app.js" defer></script><script type="module" src="/website/_assets/77c74d1469cb7f454e28-app.js" defer></script><script nomodule src="/website/_assets/legacy-88ac7487e35e949e3d72-pages/jobs/index.js" defer></script><script type="module" src="/website/_assets/f0d4bdeed546a8fa9663-pages/jobs/index.js" defer></script><script nomodule src="/website/_assets/legacy-2c7811a172afff8c6bf8-node_modules/0d43857a.js" defer></script><script type="module" src="/website/_assets/3abb1dc591b767ee0b2b-node_modules/0d43857a.js" defer></script><script nomodule src="/website/_assets/legacy-653b3056e44d5a749116-node_modules/288dc8e0.js" defer></script><script type="module" src="/website/_assets/b2fc2f85c568c6b1db7f-node_modules/288dc8e0.js" defer></script><script nomodule src="/website/_assets/legacy-183dca45315cf1c7b6cc-commons/62c6da1a.js" defer></script><script type="module" src="/website/_assets/d633932a25d845a51e42-commons/62c6da1a.js" defer></script><script nomodule src="/website/_assets/legacy-bfce6b2e40d22d445dae-pages/empty.js" defer></script><script type="module" src="/website/_assets/12bd90c4abb406a77f8c-pages/empty.js" defer></script><script src="/website/_assets/legacy-628a2edab1d8fc14b6cd-node_modules/5e2083c4.js" defer></script><script nomodule src="/website/_assets/legacy-316a95a407ef8012cc1c-node_modules/commons.js" defer></script><script type="module" src="/website/_assets/9565f2f1edecce6df008-node_modules/commons.js" defer></script><script nomodule src="/website/_assets/legacy-83681289f48116ce6529-app.js" defer></script><script type="module" src="/website/_assets/a63eb854b1bec9fe25ba-app.js" defer></script>`

the interesting part is 5e2083c4.js again .. it also has no nomodule tag, so the issue actually still exists, probably just the order of script tags is different which makes it work somehow.

**Comment by clarkdo** [comment]
Can you please go to `node_modules/@nuxt/vue-renderer/`, search `<script nomodule` and add log to print `noModuleTag + moduleTag` before `return`  ?

**Comment by clarkdo** [comment]
@simllll I may know the reason of this bug now, `5e2083c4.4f8e4af.js` is a special bundle which doesn't have corresponding modern bundle which may be because it's extracted into other common chunks as size is small or some other unknown reason.

Let's confirm my thought, if you can see `5e2083c4.4f8e4af.js` in log with my above comment, can you try to change this and try if issue is fixed ?
![image](https://user-images.githubusercontent.com/4312154/90267995-3c609780-de4e-11ea-8272-e55d15f1b54d.png)

```
// From
if (!modernJsFile) {
  return scriptTag
}
// To 
if (!modernJsFile) {
  return scriptTag.replace('<script', `<script nomodule`)
}
```

**Comment by simllll** [comment]
> Can you please go to `node_modules/@nuxt/vue-renderer/`, search `<script nomodule` and add log to print `noModuleTag + moduleTag` before `return` ?

noModuleTag + moduleTag  <script nomodule src="/website/_assets/runtime.fa009b9.js" defer></script><script type="module" src="/website/_assets/runtime.3f02afd.modern.js" defer></script>                              17:50:23
noModuleTag + moduleTag  <script nomodule src="/website/_assets/40.c765e0f.js" defer></script><script type="module" src="/website/_assets/37.c2155fc.modern.js" defer></script>                                        17:50:23
noModuleTag + moduleTag  <script nomodule src="/website/_assets/129.1dd27da.js" defer></script><script type="module" src="/website/_assets/126.7f1e543.modern.js" defer></script>                                      17:50:23
noModuleTag + moduleTag  <script nomodule src="/website/_assets/30.86aeb8b.js" defer></script><script type="module" src="/website/_assets/288.16f3f35.modern.js" defer></script>                                       17:50:23
noModuleTag + moduleTag  <script nomodule src="/website/_assets/130.be13865.js" defer></script><script type="module" src="/website/_assets/127.75b77f7.modern.js" defer></script>                                      17:50:23
noModuleTag + moduleTag  <script nomodule src="/website/_assets/pages/index.cb5a203.js" defer></script><script type="module" src="/website/_assets/pages/index.f5f2fa3.modern.js" defer></script>                      17:50:23
noModuleTag + moduleTag  <script nomodule src="/website/_assets/node_modules/0d43857a.28504fe.js" defer></script><script type="module" src="/website/_assets/node_modules/0d43857a.cf4d41c.modern.js" defer></script>  17:50:23
noModuleTag + moduleTag  <script nomodule src="/website/_assets/commons/62c6da1a.d8fb4f0.js" defer></script><script type="module" src="/website/_assets/commons/62c6da1a.de12725.modern.js" defer></script>            17:50:23
noModuleTag + moduleTag  <script nomodule src="/website/_assets/9.ef6eb2b.js" defer></script><script type="module" src="/website/_assets/8.8e919f8.modern.js" defer></script>                                          17:50:23
noModuleTag + moduleTag  <script nomodule src="/website/_assets/node_modules/1be2c077.cbf2bed.js" defer></script><script type="module" src="/website/_assets/node_modules/1be2c077.3cea03c.modern.js" defer></script>  17:50:23
noModuleTag + moduleTag  <script nomodule src="/website/_assets/17.a0c24f0.js" defer></script><script type="module" src="/website/_assets/167.746bf9a.modern.js" defer></script>                                       17:50:23
noModuleTag + moduleTag  <script nomodule src="/website/_assets/139.4d8b66b.js" defer></script><script type="module" src="/website/_assets/136.fec6980.modern.js" defer></script>                                      17:50:23
noModuleTag + moduleTag  <script nomodule src="/website/_assets/284.52d612b.js" defer></script><script type="module" src="/website/_assets/280.1006996.modern.js" defer></script>                                      17:50:23
noModuleTag + moduleTag  <script nomodule src="/website/_assets/14.9aaf416.js" defer></script><script type="module" src="/website/_assets/137.2902e3c.modern.js" defer></script>                                       17:50:23
noModuleTag + moduleTag  <script nomodule src="/website/_assets/112.258d4b4.js" defer></script><script type="module" src="/website/_assets/11.cb95906.modern.js" defer></script>                                       17:50:23
noModuleTag + moduleTag  <script nomodule src="/website/_assets/node_modules/commons.c1101f0.js" defer></script><script type="module" src="/website/_assets/node_modules/commons.b005655.modern.js" defer></script>    17:50:23
noModuleTag + moduleTag  <script nomodule src="/website/_assets/app.2417406.js" defer></script><script type="module" src="/website/_assets/app.fa2606e.modern.js" defer></script>                                      17:5

**Comment by simllll** [comment]
Alright, it's not in the output, but I guess that's what you mean, beacuse the return is done before.

I changed the return to your suggestion and IT WORKS!!! :partying_face: :partying_face: :partying_face: 

I'm really impressed how fast you tackled this issue!

Thank you very very much!

I will try to patch this for now at my side and see if it still holds in production. But I assume so! Thank you very much again!

**Comment by simllll** [comment]
Can you elaborate why there are bundles that have a legacy file, but no a modern version? This kinda confuses me.

**Comment by clarkdo** [comment]
@simllll All thanks to the useful information you provided, I've opened a pr to fix this edge case, it should be released in next patch release after code review.

/cc @pi0

**Comment by clarkdo** [comment]
> Can you elaborate why there are bundles that have a legacy file, but not a modern version? This kinda confuses me.

I think you can have a look at the content of that bundle and search which file is same content in modern bundle, if it's hard to read, you can disable terser and minimizer.

If that chunk is an async component and also imported synchrousely, webpack might bundle it into a common module if component is small in modern bundle and the parent comonent is also using that common module.

**Comment by simllll** [comment]
Some final comments from my side:
1.) I also applied the patch to my production deployment, it works perfectly!
2.) We had some issues in the past with different kind of includes (async vs direct/sync import mixed in different components), and I just tested reverting one related commit that solved an issue that we couldn't really explain in the past -> it seems this was the same root cause! This also works now as expected! Nice (y)

Thanks again, I'll close this.
